src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -179,6 +175,7 @@ int sd_dhcp_server_new(sd_dhcp_server **ret, int ifindex) {
         server->leases_by_client_id = hashmap_new(&client_id_hash_ops);
         if (!server->leases_by_client_id)
                 return -ENOMEM;
+        hashmap_set_free_value(server->leases_by_client_id, (hashmap_destroy_t) dhcp_lease_free);
```

Comment:
i wonder if we can make the cast to (hashmap_destroy_t) typesafe in some way. Maybe something like this:

```c
#define MAKE_DESTROY_FUNC(func) ({ if (false) func(NULL); (destroy_func_t) func; })
```

This way we cast to destroy_func_t but first make sure we can safely call the function with a ptr, which is all we need.